plugins/vppv2/aclplugin/descriptor/acl.go

```diff
@@ -0,0 +1,397 @@
+				err := c.aclHandler.SetMacIPACLToInterface(vppACLIndex, aclMacInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to set MAC IP ingress interfaces to ACL %s: %v",
+						newACL.AclName, err)
+				}
+			}
+		} else {
+			aclOldInInterfaces := c.getInterfaces(oldACL.Interfaces.Ingress)
+			aclOldEgInterfaces := c.getInterfaces(oldACL.Interfaces.Egress)
+			aclNewInInterfaces := c.getOrCacheInterfaces(newACL.Interfaces.Ingress, vppACLIndex, INGRESS)
+			aclNewEgInterfaces := c.getOrCacheInterfaces(newACL.Interfaces.Egress, vppACLIndex, EGRESS)
+			addedInInterfaces, removedInInterfaces := diffInterfaces(aclOldInInterfaces, aclNewInInterfaces)
+			addedEgInterfaces, removedEgInterfaces := diffInterfaces(aclOldEgInterfaces, aclNewEgInterfaces)
+
+			if len(removedInInterfaces) > 0 {
+				err := c.aclHandler.RemoveIPIngressACLFromInterfaces(vppACLIndex, removedInInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to remove IP ingress interfaces from ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+			if len(removedEgInterfaces) > 0 {
+				err := c.aclHandler.RemoveIPEgressACLFromInterfaces(vppACLIndex, removedEgInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to remove IP egress interfaces from ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+			if len(addedInInterfaces) > 0 {
+				err := c.aclHandler.SetACLToInterfacesAsIngress(vppACLIndex, addedInInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to set IP ingress interfaces to ACL %s: %v",
+						newACL.AclName, err)
+				}
+			}
+			if len(addedEgInterfaces) > 0 {
+				err := c.aclHandler.SetACLToInterfacesAsEgress(vppACLIndex, addedEgInterfaces)
+				if err != nil {
+					return errors.Errorf("failed to add IP egress interfaces to ACL %s: %v",
+						oldACL.AclName, err)
+				}
+			}
+		}*/
+	}
+
+	return
+}
+
+func (d *AclDescriptor) ModifyWithRecreate(key string, oldACL, newACL *acl.Acl, metadata *aclidx.AclMetadata) bool {
+	var hasL2 bool
+	for _, rule := range oldACL.Rules {
+		if rule.GetMatch().GetMacipRule() != nil {
+			hasL2 = true
+		} else if rule.GetMatch().GetIpRule() != nil && hasL2 {
+			return true
+		}
+	}
+	return false
+}
+
+/*func (d *AclDescriptor) Update(key string, value *acl.Acl, metadata *aclidx.AclMetadata) error {
+
+}*/
+
+func (d *AclDescriptor) Dependencies(key string, value *acl.Acl) []api.Dependency {
+	return nil
+}
+
+func (d *AclDescriptor) DerivedValues(key string, value *acl.Acl) []api.KeyValuePair {
+	return nil
+}
+
+func (d *AclDescriptor) Dump(correlate []adapter.AclKVWithMetadata) (dump []adapter.AclKVWithMetadata, err error) {
+	ipACLs, err := d.aclHandler.DumpIPACL(d.ifPlugin.GetInterfaceIndex())
+	if err != nil {
+		return nil, errors.Errorf("failed to dump IP ACLs: %v", err)
+	}
+	macipACLs, err := d.aclHandler.DumpMACIPACL(d.ifPlugin.GetInterfaceIndex())
+	if err != nil {
+		return nil, errors.Errorf("failed to dump MAC IP ACLs: %v", err)
+	}
```

Comment:
Also handle ACLs without a tag, e.g. added manually through CLI (key is going to be invalid) - IMHO we should remove stuff that was added behind the agent's back. To do that just generate name for such ACL that is unlikely going to be defined in NB, thus will get removed by resync.
For example, for untagged interfaces: https://github.com/milanlenco/vpp-agent/blob/refactor/plugins/vppv2/ifplugin/descriptor/interface_crud.go#L462-L466
In this case the name can be UNTAGGED-[counter]